### PR TITLE
Prevent accidental type literal parse

### PR DESCRIPTION
**Pull request by rick-nguyen**
Visit isn't implemented yet for TypeLiteral nodes, causing an exception in Pretty print. This is a mitigation using ParserOptions() to avoid that.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Parser/ParserOptions.cs [hunk 1]
```diff
@@ -44,16 +44,22 @@ public class ParserOptions
...
+        public ParserOptions()
+        {
+        }
+
+        public ParserOptions(CultureInfo culture, bool allowsSideEffects = false, int maxExpressionLength = 0, bool allowParseAsTypeLiteral = false)
+        {
+            Culture = culture;
+            AllowsSideEffects = allowsSideEffects;
+            MaxExpressionLength = maxExpressionLength;
+            AllowParseAsTypeLiteral = allowParseAsTypeLiteral;
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Parser/ParserOptions.cs:61]: >AllowParseAsTypeLiteral [](http://example.com/codeflow?start=12&length=23)

Does this need to be part of ctor? I don't see other flags are set this way. Default value of bool is false anyway, I think.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs [hunk 2]
```diff
@@ -71,14 +72,27 @@ private TexlParser(IReadOnlyList<Token> tokens, Flags flags, Features features =
...
+        public static ParseUDFsResult ParseUDFsScript(string script, CultureInfo loc = null, bool numberIsFloat = false)
+        {
+            Contracts.AssertValue(script);
+            Contracts.AssertValueOrNull(loc);
+
+            // UDFs always support ReservedKeywords, no back compat concern
+            var formulaTokens = TokenizeScript(script, loc, Flags.NamedFormulas | (numberIsFloat ? Flags.NumberIsFloat : 0));
+            var parser = new TexlParser(formulaTokens, Flags.NamedFormulas | (numberIsFloat ? Flags.NumberIsFloat : 0), new ParserOptions());
+
+            return parser.ParseUDFs(script);
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs:87]: Why this change?

Comment (rick-nguyen) [reply src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs]: Existing API that was removed by Anderson recently, I kept it by mistake

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs [hunk 3]
```diff
@@ -55,13 +55,14 @@ public enum Flags
...
         private readonly Features _features;
+        private readonly ParserOptions _parserOptions;
 
         // Represents temporary extra trivia, for when a parsing method
         // had to parse tailing trivia to do 1-lookahead. Will be
         // collected by the next call to ParseTrivia.
         private ITexlSource _extraTrivia;
 
-        private TexlParser(IReadOnlyList<Token> tokens, Flags flags, Features features = null)
+        private TexlParser(IReadOnlyList<Token> tokens, Flags flags, ParserOptions parserOptions, Features features = null)
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs:65]: > Flags flags, ParserOptions parserOptions [](http://example.com/codeflow?start=55&length=41)

We have two ways now which could be conflicting. Can we have a single way of defining the behavior? Can we add a flag as well for this?

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:
